python/generateTemplate.py

```diff
@@ -0,0 +1,1155 @@
+        if self.debug:
+            array_temp = array
+            array = np.zeros_like(exposure.getMaskedImage().getImage().getArray())
+            array[y0: y0 + dy, x0: x0 + dx] = array_temp
+            if variance is not None:
+                variance_temp = variance
+                variance = np.zeros_like(array)
+                variance[y0: y0 + dy, x0: x0 + dx] = variance_temp
+        exposure.getMaskedImage().getImage().getArray()[:, :] = array
+        if variance is None:
+            variance = np.abs(array)
+        exposure.getMaskedImage().getVariance().getArray()[:, :] = variance
+
+        if self.mask is not None:
+            exposure.getMaskedImage().getMask().getArray()[:, :] = self.mask
+        ha_term1 = np.sin(elevation.asRadians())
+        ha_term2 = np.sin(dec.asRadians())*np.sin(self.observatory.getLatitude().asRadians())
+        ha_term3 = np.cos(dec.asRadians())*np.cos(self.observatory.getLatitude().asRadians())
+        hour_angle = np.arccos((ha_term1 - ha_term2) / ha_term3)
+        mjd = 59000.0 + (self.observatory.getLatitude().asDegrees()/15.0 - hour_angle*180/np.pi)/24.0
+        airmass = 1.0/np.sin(elevation.asRadians())
+        if era is None:
+            era = Angle(hour_angle - self.observatory.getLongitude().asRadians())
+        meta = exposure.getMetadata()
+        meta.add("CHIPID", "R22_S11")
+        # Required! Phosim output stores the snap ID in "OUTFILE" as the last three characters in a string.
+        meta.add("OUTFILE", ("SnapId_%3.3i" % snap))
+
+        meta.add("TAI", mjd)
+        meta.add("MJD-OBS", mjd)
+
+        meta.add("EXTTYPE", "IMAGE")
+        meta.add("EXPTIME", self.exposure_time)
+        meta.add("AIRMASS", airmass)
+        meta.add("ZENITH", 90. - elevation.asDegrees())
+        meta.add("AZIMUTH", azimuth.asDegrees())
+
+        # Add all additional keyword arguments to the metadata.
+        for add_item in kwargs:
+            meta.add(add_item, kwargs[add_item])
+
+        visitInfo = afwImage.makeVisitInfo(exposureId=int(exposureId),
+                                           exposureTime=self.exposure_time,
+                                           darkTime=self.exposure_time,
+                                           date=DateTime(mjd),
+                                           ut1=mjd,
+                                           era=era,
+                                           boresightRaDec=IcrsCoord(ra, dec),
+                                           boresightAzAlt=Coord(azimuth, elevation),
+                                           boresightAirmass=airmass,
+                                           boresightRotAngle=boresightRotAngle,
+                                           observatory=self.observatory,
+                                           weather=weather
+                                           )
+        exposure.getInfo().setVisitInfo(visitInfo)
+
+        #Set the DCR-matched PSF
```

Comment:
I suspect your code could use a `flake8` linter pass, as it would complain about # without a space after it.